aws-ecr-replicationconfiguration/src/main/java/software/amazon/ecr/replicationconfiguration/DeleteHandler.java

```diff
@@ -0,0 +1,64 @@
+package software.amazon.ecr.replicationconfiguration;
+
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.services.ecr.EcrClient;
+import software.amazon.awssdk.services.ecr.model.PutReplicationConfigurationRequest;
+import software.amazon.awssdk.services.ecr.model.ReplicationConfiguration;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+import java.util.Collections;
+
+public class DeleteHandler extends BaseHandlerStd {
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<EcrClient> proxyClient,
+        final Logger logger) {
+
+        final ResourceModel model = request.getDesiredResourceState();
+        final EcrClient client = proxyClient.client();
+
+        try {
+            // Ensure resource exists on the ECR Registry.
```

Comment:
What happens when the replication configuration has been deleted manually and then a customer updates their stack to remove this resource? I think this would make it fail, and I'm not sure that's what we'd want.

Reply:
This is part of the CloudFormation contract. 

"A delete hander MUST return FAILED with a NotFound error code if the if the resource did not exist prior to the delete request."

https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html